### PR TITLE
Adds Mouse Text Selection functionality for chromebook

### LINKED ISSUE #15754: Text selection does not work with a mouse
The user should be able to press the mouse button, drag over text to select, then release.

### PR DESCRIPTION
**Pull request by jslavitz**
Fixes #15754
-Needs monodrag fix to land before this will pass tests, but I wanted to get the review started.

### CODE REVIEW

File: packages/flutter/lib/src/gestures/drag_details.dart [hunk 1]
```diff
@@ -20,14 +20,17 @@ class DragDownDetails {
   /// Creates details for a [GestureDragDownCallback].
   ///
   /// The [globalPosition] argument must not be null.
-  DragDownDetails({ this.globalPosition = Offset.zero })
+  DragDownDetails({ this.globalPosition = Offset.zero, this.deviceKind = PointerDeviceKind.touch})
```
Comment (Hixie) [inline packages/flutter/lib/src/gestures/drag_details.dart]: nit: please make sure to be consistent in the style. so for example here, you have a space after the `{` so you should have a space before the `}`.

**Comment by Hixie** [comment]
Can you elaborate on why we would want the mouseSlop and enableForMouse features? Is this based on a UX design doc?

**Comment by jslavitz** [comment]
So I just made both of those up myself. I thought that the mouse slop would be useful in the case there is a slight jittering of the mouse when the user moved it though I'm not 100% sure if it's necessary. The enableForMouse feature is, however needed, so that developers can program features that need the mouse to be able to perform drag actions. I only set that variable to false for the scrollable widget.

**Comment by Hixie** [comment]
Why wouldn't drag features just work automatically, like with touch?

I really think we shouldn't be designing UI here. We should get clear guidance from UX and make sure we implement what they want mouse input to do.

**Comment by jslavitz** [comment]
They do? It's just because we have scrolling on a mouse, we need to disable drag scrolling for the mouse. The boolean is only meant to be used in this one situation where we are disabling a feature specifically because it's implemented differently for the mouse (ie. scrolling). This allows things to work the same otherwise between mouse and touch.

**Comment by Hixie** [comment]
I don't really follow. Maybe you can catch me at the office and show me a demo of what the differences are. :-)

**Comment by Hixie** [comment]
@jslavitz Given the more recent UX guidance regarding mouse dragging causing scrolling, we should probably close this PR and approach this afresh. How does that sounds to you?

**Pull request closed without merging**